src/pages/base.js

```diff
@@ -16,8 +16,18 @@ class Page {
   visitWithParameters(data) {
     let url = this.url;
 
+    const additionalParams = [];
     for (const item of data.raw()) {
-      url = url.replace(`:${item[0]}`, item[1]);
+      if (url.indexOf(`:${item[0]}`) === -1) {
+        additionalParams.push(item);
+      } else {
+        url = url.replace(`:${item[0]}`, item[1]);
+      }
+    }
+
+    if (additionalParams.length > 0) {
+      const esc = encodeURIComponent;
+      url = url + '?' + additionalParams.map(item => esc(item[0]) + '=' + esc(item[1])).join('&');
```

Comment:
I think that we shouldn't overwrite the existing `url` variable.

Maybe it would be good to use reduce instead of this loop, so you can pass this.url as an initial value

and then create a method e.g. `const transformedUrl` which will be passed to the `browser.get(transformedUrl)` method. 
This method should be responsible for building the new `transformedUrl`. 

What do you think? @sethii

Reply:
I've refactored visitWithParams, although IMO now it looks less readdable.

Reply:
Indeed, I thought to make it a bit less functional. Just assign `data.reduce()` to other variable. 

But let's wait for @sethii's opinion